sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/ShareClientBuilder.java

```diff
@@ -189,8 +190,9 @@ public ShareClientBuilder endpoint(String endpoint) {
             }
             this.shareName = length >= 2 ? pathSegments[1] : this.shareName;
 
+            // TODO : What happens if a user has custom queries?
```

Comment:
This is a good question that has gone unanswered for now, @rickle-msft thoughts? These could just be thrown back into the query string when the SAS token is generated, that way we are able to clean the SAS token using the SAS query parameters classes but don't lose custom query parameters unrelated to SAS tokens.

Reply:
Created an issue here https://github.com/Azure/azure-sdk-for-java/issues/6604